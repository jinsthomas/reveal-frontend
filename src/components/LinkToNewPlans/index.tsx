import * as React from 'react';
import { Link } from 'react-router-dom';
import { NEW_PLAN_URL } from '../../constants';

export interface Props {
  classProp: string;
}

const LinkToNewPlans = (props: Props) => {
  return (
    <Link to={NEW_PLAN_URL} className={props.classProp}>
      Add Focus Investigation
    </Link>
  );
};

const defaultProps: Props = { classProp: 'focus-investigation btn btn-primary float-right mt-5' };
LinkToNewPlans.defaultProps = defaultProps;

export default LinkToNewPlans;
