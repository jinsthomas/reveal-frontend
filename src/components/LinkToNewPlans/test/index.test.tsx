import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Router } from 'react-router';
import LinkToNewPlans from '..';

const history = createBrowserHistory();

describe('components/LinkToNewPlans', () => {
  it('renders without crashing', () => {
    shallow(<LinkToNewPlans />);
  });

  it('Matches snapshot', () => {
    const wrapper = mount(
      <Router history={history}>
        <LinkToNewPlans />
      </Router>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
