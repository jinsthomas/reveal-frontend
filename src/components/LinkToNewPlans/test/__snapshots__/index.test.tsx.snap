// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LinkToNewPlans Matches snapshot 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <LinkToNewPlans
    classProp="focus-investigation btn btn-primary float-right mt-5"
  >
    <Link
      className="focus-investigation btn btn-primary float-right mt-5"
      replace={false}
      to="/plans/new"
    >
      <a
        className="focus-investigation btn btn-primary float-right mt-5"
        href="/plans/new"
        onClick={[Function]}
      >
        Add Focus Investigation
      </a>
    </Link>
  </LinkToNewPlans>
</Router>
`;
